vars/mavenFunctionalTest.groovy

```diff
@@ -41,17 +50,17 @@ def call(body) {
             def deployEnv = []
             if (config.testVaultTokenRole != null) {
                 stage('Request Vault Token for testing') {
-                     withCredentials([string(credentialsId: 'jenkins-vault', variable: 'JENKINS_VAULT_TOKEN')]) {
-                        vaultToken = sh(returnStdout: true, script: "curl -k -s --header \"X-Vault-Token: ${JENKINS_VAULT_TOKEN}\" --request POST --data '{\"display_name\": \"testenv\"}' ${env.VAULT_ADDR}/v1/auth/token/create/${config.testVaultTokenRole}?ttl=30m | jq '.auth.client_token'").trim().replaceAll('"', '')
+                     withCredentials([string(credentialsId: "${config.vaultCredID}", variable: 'JENKINS_VAULT_TOKEN')]) {
+                        vaultToken = sh(returnStdout: true, script: "curl -k -s --header \"X-Vault-Token: ${JENKINS_VAULT_TOKEN}\" --request POST --data '{\"display_name\": \"testenv\"}' ${config.vaultAddr}/v1/auth/token/create/${config.testVaultTokenRole}?ttl=30m | jq '.auth.client_token'").trim().replaceAll('"', '')
                         deployEnv.add("VAULT_TOKEN=${vaultToken}")
                     }
-                } 
+                }
             }
 
             stage('Functional Testing') {
                 echo "Executing functional tests against ${config.serviceUrl}"
                 withEnv(deployEnv) {
-                    sh "${mvnCmd} integration-test -P inttest -Dbrowser=HtmlUnit -Dtest.env=ci -DbaseURL=${config.serviceUrl} -Djavax.net.ssl.keyStore=${config.keystore} -Djavax.net.ssl.keyStorePassword=${config.keystorePassword} -Djavax.net.ssl.trustStore=/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/security/cacerts -Djavax.net.ssl.trustStorePassword=changeit -DX-Vault-Token=${env.VAULT_TOKEN} -Dvault.url.domain='${env.VAULT_ADDR}' -Dcucumber.options='${config.cucumberOpts}' ${config.options}"
+                    sh "${mvnCmd} integration-test -P inttest -Dbrowser=HtmlUnit -Dtest.env=ci -DbaseURL=${config.serviceUrl} -Djavax.net.ssl.keyStore=${config.keystore} -Djavax.net.ssl.keyStorePassword=${config.keystorePassword} -Djavax.net.ssl.trustStore=/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/security/cacerts -Djavax.net.ssl.trustStorePassword=changeit -DX-Vault-Token=${config.vaultToken} -Dvault.url.domain='${config.vaultAddr}' -Dcucumber.options='${config.cucumberOpts}' ${config.options}"
```

Comment:
So the intent for this vault token is that its value is the token returned by the vault call on line #54.

Reply:
This looks like a bug that existed in the library already.